database/staking.go

```diff
@@ -439,3 +447,37 @@ func (db BigDipperDb) SaveRedelegations(redelegations []types.Redelegation) erro
 	_, err = db.Sql.Exec(redelQuery, redelParams...)
 	return err
 }
+
+//SaveDelegationsShare sve an array of delegation share
+func (db BigDipperDb) SaveDelegationsShares(shares []types.DelegationShare) error {
+	stmt := `INSERT INTO validator_delegation_shares (operator_address ,delegator_address,shares,height,timestamp) VALUES`
+	var delegationShareParam []interface{}
+	for i, share := range shares {
+		i1 := i * 5
+		stmt += fmt.Sprintf("($%d,$%d,$%d,$%d,$%d),", i1+1, i1+2, i1+3, i1+4, i1+5)
+		delegationShareParam = append(delegationShareParam, share.ValidatorAddress.String(), share.DelegatorAddress.String(), share.Shares, share.Height, share.Timestamp)
+	}
+	stmt = stmt[:len(stmt)-1] // Remove the trailing ","
+	stmt += " ON CONFLICT DO NOTHING"
+	_, err := db.Sql.Exec(stmt, delegationShareParam...)
+	if err != nil {
+		return err
+	}
+
+	//insert self delegation in this entry into self_deleation_row
+	//for each time there are someone delegated to that validator, save the validator_self delegation?
+	stmt = `INSERT INTO validator_self_delegation (operator_address ,shares,height,timestamp)
+	SELECT validator_delegation_shares.operator_address, shares,height,timestamp
+	FROM validator_delegation_shares INNER JOIN validator_info
+	ON (validator_delegation_shares.delegator_address = validator_info.self_delegate_address
+		AND validator_delegation_shares.operator_address = validator_info.operator_address)
+	WHERE height = $1`
+
+	//Since shares have same timestamp
+	_, err = db.Sql.Exec(stmt, shares[0].Height)
```

Comment:
Why are we using this table? Couldn't we simply use `validator_delegation_shares` and search for `delegator_address == validator.self_delegation_address`?

Reply:
I think this is a more clear and easier way to do more further calculations, like self delegation ratio.
If that is not nesserary we can just delete that.

Reply:
@HarleyAppleChoi I think it's fine deleting this, since thanks to the GraphQL APIs that will be built on top of this database, we will have that data in any case. So we can avoid duplicates